src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs

```diff
@@ -56,8 +56,320 @@ private static unsafe void ThrowExceptionEncryptDecryptFail(string fullPath)
             throw Win32Marshal.GetExceptionForWin32Error(errorCode, fullPath);
         }
 
+        private static unsafe bool TryCloneFile(string sourceFullPath, string destFullPath, bool overwrite)
+        {
+            // Open the source file.
+            // We use FILE_FLAGS_NO_BUFFERING since we're not using unaligned writes during cloning and can skip buffering overhead.
+            const int FILE_FLAG_NO_BUFFERING = 0x20000000;
+            using (SafeFileHandle sourceHandle = Interop.Kernel32.CreateFile(
+                sourceFullPath,
+                Interop.Kernel32.GenericOperations.GENERIC_READ,
+                FileShare.Read,
+                FileMode.Open,
+                FILE_FLAG_NO_BUFFERING))
+            {
+                // Return false if we failed to open the source file.
+                if (sourceHandle.IsInvalid)
+                {
+                    return false;
+                }
+
+                // Read the source's volume's info.
+                uint sourceSerialNumber;
+                uint sourceVolumeFlags;
+                if (!Interop.Kernel32.GetVolumeInformationByHandle(sourceHandle, null, 0, &sourceSerialNumber, null, &sourceVolumeFlags, null, 0))
+                {
+                    throw new Exception("C");
+                    //return false;
```

Comment:
For the final code where you might want to trace the result (or why it could not clone) consider changing return type to `string?` where null means cloned successfully, non-null means clone failure, and the non-null string is the operation that failed.